src/Tizen.NUI.WindowSystem/src/public/SoftkeyClient.cs

```diff
@@ -0,0 +1,336 @@
+            {
+                Dispose(DisposeTypes.Explicit);
+                GC.SuppressFinalize(this);
+            }
+        }
+
+        /// <inheritdoc/>
+        protected virtual void Dispose(DisposeTypes type)
+        {
+            int res;
+            if (!disposed)
+            {
+                if (_softkeyClient != IntPtr.Zero)
+                {
+                    res = Interop.SoftkeyClient.Destroy(_softkeyClient);
+                    try
+                    {
+                        _tzsh.ErrorCodeThrow(res);
+                    }
+                    catch (ArgumentException e)
+                    {
+                        throw new MemberAccessException("SoftkeyClient is a corrupted");
+                    }
+                    _softkeyClient = IntPtr.Zero;
+                }
+                disposed = true;
+            }
+        }
+
+        /// <summary>
+        /// Requests to show the softkey service window.
+        /// </summary>
+        /// <exception cref="ArgumentException">Thrown when failed of invalid argument.</exception>
+        /// <exception cref="NotSupportedException">Thrown when the feature is not supported.</exception>
+        /// <exception cref="InvalidOperationException">Thrown when failed because of an invalid operation or no service.</exception>
+        /// <since_tizen> 8 </since_tizen>
+        public void Show()
+        {
+            int res;
+
+            res = Interop.SoftkeyClient.Show(_softkeyClient);
+            _tzsh.ErrorCodeThrow(res);
+        }
+
+        /// <summary>
+        /// Requests to hide the softkey service window.
+        /// </summary>
+        /// <exception cref="ArgumentException" > Thrown when failed of invalid argument.</exception>
+        /// <exception cref="NotSupportedException">Thrown when the feature is not supported.</exception>
+        /// <exception cref="InvalidOperationException">Thrown when failed because of an invalid operation or no service.</exception>
+        /// <since_tizen> 8 </since_tizen>
+        public void Hide()
+        {
+            int res;
+
+            res = Interop.SoftkeyClient.Hide(_softkeyClient);
+            _tzsh.ErrorCodeThrow(res);
+        }
+
+        /// <summary>
+        /// Gets the visible state of a softkey service window.
+        /// </summary>
+        /// <returns>The visible state of the softkey service window.</returns>
+        /// <exception cref="ArgumentException" > Thrown when failed of invalid argument.</exception>
+        /// <exception cref="NotSupportedException">Thrown when the feature is not supported.</exception>
+        /// <exception cref="InvalidOperationException">Thrown when failed because of an invalid operation or no service.</exception>
+        /// <since_tizen> 8 </since_tizen>
+        public VisibleState Visible
+        {
+            get
+            {
+                return GetVisible();
+            }
+        }
+
+        private VisibleState GetVisible()
```

Comment:
Keep styling rules for ordering of class members : see https://stackoverflow.com/questions/150479/order-of-items-in-classes-fields-properties-constructors-methods

Reply:
I re-order whole method, property, event and enum in classes